Application.cpp

```diff
@@ -319,7 +321,7 @@ bool Application::onInit() {
 	// Create the pipeline layout
 	PipelineLayoutDescriptor layoutDesc{};
 	layoutDesc.bindGroupLayoutCount = 1;
-	layoutDesc.bindGroupLayouts = &(WGPUBindGroupLayout)bindGroupLayout;
+	layoutDesc.bindGroupLayouts = &static_cast<WGPUBindGroupLayout&>(bindGroupLayout);
```

Comment:
Didn't `(WGPUBindGroupLayout*)&bindGroupLayout` work? Doesn't your compiler still warn about the use of pointer to a rvalue? Dereferencing first, then casting the pointer seems safer to me.